src/ape_logging.c

```diff
@@ -0,0 +1,66 @@
+/*
+   Copyright 2016 Nidium Inc. All rights reserved.
+   Use of this source code is governed by a MIT license
+   that can be found in the LICENSE file.
+*/
+
+#include <string.h>
+#include <stdio.h>
+
+#include "ape_logging.h"
+
+
+void APE_SetLogger(ape_logger_t * logger, const ape_log_lvl_t lvl, \
+    const ape_log_init_cbt init, const ape_log_log_cbt log,
+    const ape_log_clear_cbt clear, void * cb_args)
+{
+        if (logger->clear) {
+            logger->clear(logger->cb_args);
+            logger->cb_args = NULL;
+        }
+        logger->lvl = lvl;
+        logger->init = init;
+        logger->log = log;
+        logger->clear = clear;
+        logger->cb_args = cb_args;
+        if (logger->init) {
+            logger->init(logger->cb_args);
+        }
+}
+
+
+// keep these in sync wit ape_logging.h: _ape_log_lvl_t
+static const char * ape_log_levellabels[] = {"DEBUG", "WARN", "ERROR", "INFO"};
```

Comment:
I'd like to keep this is this file because it is only used here.

This static variable (not a define)  is only needed to inside the 'APE_Log' function.
There it is used to lookup the appriate text for the 'ape_log_lvl_t' enum values.
They are passed to the userdefined logger callback, so that no lookup is needed there.